idl.md

```diff
@@ -1,61 +1,90 @@
-## Proposed IDL
-```javascript
+# Proposed IDL
+
+```js
 partial interface ServiceWorkerRegistration {
-  readonly attribute SyncManager syncManager;
+  readonly attribute SyncManager sync;
+  readonly attribute PeriodicSyncManager periodicSync;
 };
 
 interface SyncManager {
   Promise<SyncRegistration> register(optional SyncRegistrationOptions options);
-  Promise<SyncRegistration> getRegistration(DOMString id);
+  Promise<SyncRegistration> getRegistration(DOMString tag);
   Promise<sequence<SyncRegistration>> getRegistrations();
-  Promise<SyncPermissionStatus> hasPermission();
-  readonly attribute unsigned long minAllowablePeriod;
+  Promise<SyncPermissionState> permissionState();
```

Comment:
One more thing I was wondering about, should permissionState be an attribute (still returning the same Promise) rather than a method? Isn't it similar in nature to Service Worker's .ready and CSS Font Loading's .ready?

Re vs. hasPermission(): overall, I think "permissionState" wins because hasPermission() implies a boolean, isn't consistent with Permissions API. Making it an attribute would solve the nitpick about the missing "get".